AjaxControlToolkit/Scripts/AjaxFileUpload.js

```diff
@@ -636,15 +639,22 @@ Sys.Extended.UI.AjaxFileUpload.ProcessorHtml5 = function(control, elements) {
 
             // validate it, add it if it's OK
             if(control.fileTypeIsValid(fileItem.type)) {
-                if(!control.addFileToQueue(fileItem))
+                if (!control.addFileToQueue(fileItem)) {
+                    startUpload = false;
                     break;
+                }
             } else {
+                startUpload = false;
                 control.confirmFileIsInvalid(fileItem);
             }
         }
 
         // reset input file value so 'change' event can be fired again with same file name.
         elements.inputFile.value = null;
+
+        if (fileCount > 0 && startUpload && control.get_autoStartUpload()) {
+            control.startUpload();
```

Comment:
Move this check and `control.startUpload()` outside the method (see comment below).